inject-kotlin/src/main/kotlin/io/micronaut/kotlin/processing/visitor/KotlinClassElement.kt

```diff
@@ -0,0 +1,873 @@
+                    it.contains(ElementModifier.PUBLIC)
+                } else {
+                    !it.contains(ElementModifier.PRIVATE)
+                }
+            }.annotated { prop ->
+                if(prop.hasAnnotation(JvmField::class.java)) {
+                    false
+                } else {
+                    val excludedAnnotations = propertyElementQuery.excludedAnnotations
+                    excludedAnnotations.isEmpty() || !excludedAnnotations.any { prop.hasAnnotation(it) }
+                }
+            }
+
+        val allProperties : MutableList<PropertyElement> = mutableListOf()
+        // unfortunate hack since these are not excluded?
+        if (hasDeclaredStereotype(ConfigurationReader::class.java)) {
+            val configurationBuilderQuery = ElementQuery.of(PropertyElement::class.java)
+                .annotated { it.hasDeclaredAnnotation(ConfigurationBuilder::class.java) }
+                .onlyInstance()
+            val configBuilderProps = enclosedElementsQuery.getEnclosedElements(this, configurationBuilderQuery)
+            allProperties.addAll(configBuilderProps)
+        }
+
+        allProperties.addAll(enclosedElementsQuery.getEnclosedElements(this, eq))
+        val propertyNames = allProperties.map { it.name }.toSet()
+
+        val resolvedProperties : MutableList<PropertyElement> = mutableListOf()
+        val methodProperties = AstBeanPropertiesUtils.resolveBeanProperties(propertyElementQuery,
+            this,
+            {
+                getEnclosedElements(
+                    ElementQuery.ALL_METHODS
+                )
+            },
+            {
+                emptyList()
+            },
+            false, propertyNames,
+            customReaderPropertyNameResolver,
+            customWriterPropertyNameResolver,
+            { value: AstBeanPropertiesUtils.BeanPropertyData ->
+                if (!value.isExcluded) {
+                    this.mapToPropertyElement(
+                        value
+                    )
+                } else {
+                    null
+                }
+            })
+        resolvedProperties.addAll(methodProperties)
+        resolvedProperties.addAll(allProperties)
+        return resolvedProperties
+    }
+
+    private fun mapToPropertyElement(value: AstBeanPropertiesUtils.BeanPropertyData): KotlinPropertyElement {
+        return KotlinPropertyElement(
+            this@KotlinClassElement,
+            value.type,
+            value.propertyName,
+            value.field,
+            value.getter,
+            value.setter,
+            elementAnnotationMetadataFactory,
+            visitorContext,
+            value.isExcluded
+        )
+    }
+
+    override fun getSimpleName(): String {
+        var parentDeclaration = classDeclaration.parentDeclaration
+        return if (parentDeclaration == null) {
+            if (classDeclaration.qualifiedName?.asString() == Any::class.java.name)
```

Comment:
is there a test for this? i think `Any::class.java.name` might just be `java.lang.Object`, while `classDeclaration.qualifiedName` might be `kotlin.Any` or `java.lang.Object`, not sure

Reply:
ah the test caught it :laughing: